### ISSUE TITLE
Julia 1.9 does not have _maybetail in Base

### ISSUE THREAD

**Issue by jaemolihm** [JuliaApproximation/QuasiArrays.jl#85]
`_maybetail` is used twice: https://github.com/JuliaApproximation/QuasiArrays.jl/blob/b93bc59df4493c1585322989250d5914f5ff4abe/src/indices.jl#L113-L118

There is a related fix in a different package https://github.com/JuliaArrays/EndpointRanges.jl/pull/25

**Comment by dlfivefifty** [comment]
Thanks for this. I always wait until RC stage before making changes in my packages (as sometimes changes on master end up being reverted)

**Comment by jagot** [comment]
Bump :)

I've tried replacing `_maybetail(inds)` by `inds[2:end]`, as suggested here: https://github.com/JuliaLang/julia/issues/23822#issuecomment-623703610, here and in ContinuumArrays.jl.

Additionally, it seems the call signature for `uncolon` was changed with [this PR](https://github.com/JuliaLang/julia/pull/45672), which affects `to_indices`; dropping the second argument seems to work.

However, I'm unsure how to proceed to retain backwards compatibility with Julia < 1.9. I assume that `indices.jl` contains pieces basically carbon-copied from `base/multidimensional.jl`. I guess we could have a version guard in `indices.jl`, but that seems inelegant somehow.
